src/main.ts

```diff
@@ -21,4 +49,133 @@ async function run() {
   }
 }
 
+async function handleMain(inputs: Inputs, client: GitHubClient): Promise<void> {
+  const changes = await getChangesInPush();
+
+  if (changes.additions.includes(inputs.haltFile)) {
```

Comment:
I'm curious if adding/removing are the ideal semantics. I like the idea of having to add a status as the mechanism that halts the line. So you always have a `haltFile` and empty indicates green. That gives you the further niceties of easily checking the `git log` for the halt history.

The down side is that "empty" can be a funny semantic to handle. Is just whitespace empty? Does it really need to be a zero byte file? What about a BOM? So I can see where adding/removing is simpler.

Reply:
> I like the idea of having to add a status as the mechanism that halts the line

I'm not sure if you're saying "having to add a status as the mechanism that halts the line" is a feature of this as implemented or what you're suggesting. Can you clarify?

> That gives you the further niceties of easily checking the `git log` for the halt history.

I guess you mean you can do the more natural `git log -- {status file}`, instead of whatever command will still show log on a not-currently-present file (which I have to look up every time). Is that right?

> The down side is that "empty" can be a funny semantic to handle. Is just whitespace empty? Does it really need to be a zero byte file? What about a BOM? So I can see where adding/removing is simpler.

I'm still in favor of add/remove, but I think you might be over thinking it a little bit; `.contents.trim() === ""` should work.

Alternatively, you could make it something like a `halt.yaml` and require real structure, with a `halted: true|false` in there.

Reply:
Circling back, the add/remove seems simple enough for now.